### PR TITLE
[16.0][IMP] sale_stock_available_to_promise_release: show/hide availability on SO reports

### PR DESCRIPTION
**Pull request by victorvermot**
Add parameter to make the column Available on SO reports invisible

Results
- Setting enabled
OCAAB
- With setting activated
OCAAB1
- Setting disabled
OCAAB2

### CODE REVIEW

File: sale_stock_available_to_promise_release/models/res_config_settings.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Copyright 2024 Camptocamp
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    display_is_available = fields.Boolean(
```
Comment (santostelmo) [inline sale_stock_available_to_promise_release/models/res_config_settings.py]: IMO This should be true by default and related to the company. You maybe want to show this field only in some companies

File: sale_stock_available_to_promise_release/reports/sale_order.xml [hunk 2]
```diff
@@ -9,15 +9,15 @@
     >
       <xpath expr="//th[@name='th_description']" position="after">
         <th
-                t-if="doc.state != 'draft'"
+                t-if="doc.state != 'draft' and doc.env['ir.config_parameter'].sudo().get_param('sale_stock_available_to_promise_release.display_is_available')"
```
Comment (santostelmo) [inline sale_stock_available_to_promise_release/reports/sale_order.xml]:
```suggestion
                t-if="doc.state != 'draft' and doc.company_id.display_is_available"
```
As mentioned above if you configure by company then you can do something like that.

File: sale_stock_available_to_promise_release/models/res_config_settings.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        related="company_id.display_is_available",
+        help="Show Available column in SO and quotation reports",
+        readonly=False,
+        config_parameter="sale_stock_available_to_promise_release.display_is_available",
```
Comment (santostelmo) [inline sale_stock_available_to_promise_release/models/res_config_settings.py]: Do you need config parameter ?

**Review by cyrilmanuel — APPROVED** [review]

**Review by santostelmo — APPROVED** [review]

File: sale_stock_available_to_promise_release/models/res_config_settings.py (hunk 3 above)
Comment (victorvermot) [reply sale_stock_available_to_promise_release/models/res_config_settings.py]: you're right I removed it

**Review by leemannd — APPROVED** [review]

File: sale_stock_available_to_promise_release/models/res_company.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class ResCompany(models.Model):
+
+    _inherit = "res.company"
+
+    display_is_available = fields.Boolean(
```
Comment (sebalix) [inline sale_stock_available_to_promise_release/models/res_company.py]: The name of this field is too generic for `res.company` and could conflict with another field name easily with all modules installed on an Odoo instance.
What about `sale_stock_report_display_is_available` or `stock_sale_report_display_is_available` (as all other settings are prefixed with `stock_` in the parent module).

File: sale_stock_available_to_promise_release/models/res_config_settings.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    display_is_available = fields.Boolean(
+        related="company_id.display_is_available",
+        help="Show Available column in SO and quotation reports",
```
Comment (sebalix) [inline sale_stock_available_to_promise_release/models/res_config_settings.py]: I think you can remove the `help` here, it'll come from the original field automatically.

File: sale_stock_available_to_promise_release/views/res_config_settings.xml [hunk 6]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <record id="res_config_settings_view" model="ir.ui.view">
+        <field name="model">res.config.settings</field>
+        <field
+            name="inherit_id"
+            ref="stock.res_config_settings_view_form"
+        />
+        <field name="arch" type="xml">
+            <xpath expr="//div[@id='annual_inventory_date']" position="after">
```
Comment (sebalix) [inline sale_stock_available_to_promise_release/views/res_config_settings.xml]: There is already a section introduced by the parent module to host settings related to the release features: https://github.com/OCA/wms/blob/16.0/stock_available_to_promise_release/views/res_config_settings.xml#L13C64-L13C81
Better to place the new setting in this section.

**Review by sebalix — CHANGES_REQUESTED** [review]

**Review by sebalix — APPROVED** [review]

**Review by jbaudoux — APPROVED** [review]

File: sale_stock_available_to_promise_release/models/res_company.py [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+from odoo import fields, models
+
+
+class ResCompany(models.Model):
+
+    _inherit = "res.company"
+
+    stock_sale_report_display_is_available = fields.Boolean(
+        help="Show Available column in SO and quotation reports",
+        default=True,
```
Comment (jbaudoux) [inline sale_stock_available_to_promise_release/models/res_company.py:12]: Shouldn't this be `False` by default ?

cc @sebalix

Comment (sebalix) [reply sale_stock_available_to_promise_release/models/res_company.py:12]: Before this PR the column was always displayed, we are only adding an option here to hide it, but by default we keep it displayed as before. No?

Comment (jbaudoux) [reply sale_stock_available_to_promise_release/models/res_company.py:12]: all good then :)

**Comment by jbaudoux** [comment]
@victorvermot Can you drop `[16.0][ADD]: ` from the commit message ?

**Pull request merged**